docs/DeveloperGuide.md

```diff
@@ -238,33 +238,84 @@ _{more aspects and alternatives to be added}_
 
-![Clone1](images/Clone1.png)
+![CloneActivityDiagram0](images/CloneActivityDiagram0.png)
 
-The following activity diagram summarizes what happens when a user executes a new command:
+After all the checks on the index have been done, cloneCommand then calls `CloneCommand#clonePerson`, which separates
+the Person name (as a string) into two substrings using `CloneCommand#splitStringAtLastSpace`, the name and the possible
+suffix, respectfully.
+
+After being separated into two substrings, `CloneCommand#clonePerson` then separates the person name into three cases:
+Firstly, if the name has no spaces and no suffix. Secondly, if the name has spaces but the suffix does not consist of
+only an integer. And lastly, if the name has spaces and the suffix consists of only an integer. As John has no spaces
+and no suffix, this would be the first case highlighted.
+
+Based on these three cases, either than have a suffix or they do not. If they do not have a suffix, such as in the first
+two cases, then they have a suffix of one added to the back of the person name. If they name already possesses a suffix,
+then this suffix is incremented. The newly cloned person with their new name is now returned to `CloneCommand#execute`.
+As John has no suffix, a suffix of one is added to the back of his name and John 1 is returned. 
+
+![CloneActivityDiagram1](images/CloneActivityDiagram1.png)
+
+Once returned, the command then attempts to add the cloned person into the list through `Model#addPerson`. However, if
+it throws a duplicate exception (implying that a person with that name already exists in the contact book), then this
+cloned person is cloned yet again via `CloneCommand#clonePerson` until the suffix of the cloned person returned is
+unique. As the only other person in the list prior was James, John 1 is added to the list successfully on the first
+attempt to add him.
+
+![CloneActivityDiagram2](images/CloneActivityDiagram2.png)
+
+![Clone1](images/Clone1.png)
 
-![CloneActivityDiagram](images/CloneActivityDiagram.png)
+Should the person be added successfully to the address book, the clone success message will be returned to the user, as 
+depicted in the User Guide.
 
 #### Design considerations:
 
 **Aspect: How clone executes:**
 
-* **Alternative 1 (current choice):** Copies the person at the index provided and returns a person with a number next to their name
-  * Pros: Fast, prevents excessive copying of a person while ensuring that there are no struct duplicates
-  * Cons: Can be restrictive and time consuming, as you have to keep copying the clone if you wish to make a clone of a person
+* **Alternative 1 (current choice):** Copies the person at the index provided and returns a person with a number 
+    next to their name
+  * Pros: Fast,  while ensuring that there are no strict duplicates
+  * Cons: Can be restrictive as you might have contacts that are similar and have the same name.
```

Comment:
For this portion, slight inconsistency with whether there is a full-stop at the end or not.

"Pros" portion lacks a full-stop, but "Cons" portion has one